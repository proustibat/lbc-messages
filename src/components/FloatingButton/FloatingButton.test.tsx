import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import { FloatingButton, FloatingButtonProps, IconVariant } from './index';

const props: FloatingButtonProps = {
  onClick: jest.fn(),
  icon: IconVariant.ADD,
};
describe('ButtonPlus', () => {
  afterEach(() => {
    cleanup();
    jest.resetAllMocks();
  });

  it('should render correctly with add variant', () => {
    // Given / When
    const { container } = render(<FloatingButton {...props} />);

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with home variant', () => {
    // Given / When
    const { container } = render(
      <FloatingButton {...props} icon={IconVariant.HOME} />,
    );

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should handle click', () => {
    // Given
    const { getByTestId } = render(<FloatingButton {...props} />);

    // When
    fireEvent.click(getByTestId('floating-button'));

    // Then
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });

  it('should disable the click action', () => {
    // Given
    const { getByTestId } = render(
      <FloatingButton {...props} disabled={true} />,
    );

    // When
    fireEvent.click(getByTestId('floating-button'));

    // Then
    expect(props.onClick).not.toHaveBeenCalled();
  });
});
