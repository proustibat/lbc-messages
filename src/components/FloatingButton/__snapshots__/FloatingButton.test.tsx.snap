// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPlus should render correctly with add variant 1`] = `
<div>
  <button
    aria-label="add"
    class="MuiButtonBase-root MuiFab-root root_backgroundColor-#3f51b5 root_color-#fff"
    data-testid="floating-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`ButtonPlus should render correctly with home variant 1`] = `
<div>
  <button
    aria-label="home"
    class="MuiButtonBase-root MuiFab-root root_backgroundColor-#3f51b5 root_color-#fff"
    data-testid="floating-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiFab-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
